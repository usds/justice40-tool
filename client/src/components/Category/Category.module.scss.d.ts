declare namespace CategoryNamespace {
    export interface ICategoryScss {
        categoryContainer: string;
        category:string;
        disCategoryContainer: string;
    }
  }

declare const CategoryScssModule: CategoryNamespace.ICategoryScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: CategoryNamespace.ICategoryScss;
  };

  export = CategoryScssModule;
