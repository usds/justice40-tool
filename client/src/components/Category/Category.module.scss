@use '../../styles/design-system.scss' as *;

.categoryContainer {
    display: flex;
    justify-content: space-between;

    .category {
        flex-basis: 80%;
    }
}

.disCategoryContainer {
    display: flex;
    justify-content: space-between;

    .category {
        flex-basis: 80%;
    }
}
