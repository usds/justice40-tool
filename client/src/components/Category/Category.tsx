import React from 'react';
import DisadvantageDot from '../DisadvantageDot';

import * as styles from './Category.module.scss';

interface ICategory {
    name: string;
    isDisadvantaged: boolean | null;
}

/**
 * This component controls the Categories on the side panel.
 *
 * The category will be styled differently differently depending on
 * if the category is disadvantaged or not. The JSX in the return
 * statement is identical however in the global CSS file, we
 * override the disadvantaged case with a psuedo-selector (:has) that
 * is new. In order to fallback gracefully for browsers that do
 * not yet support the ":has" psuedo selector, this redundant JSX
 * will allow the disadvantaged case show the older category styling
 * while browsers that do support the ":has" psuedo selector will
 * render the newer category style.
 *
 * @param {string} name
 * @param {boolean} isDisadvagtaged
 * @return {JSX.Element}
 */
const Category = ({name, isDisadvantaged}:ICategory) => {
  return isDisadvantaged ? (
    <div className={styles.disCategoryContainer}>
      <div className={styles.category}>
        {name}
      </div>
      <DisadvantageDot isDisadvantaged={isDisadvantaged}/>
    </div>
  ) : (
    <div className={styles.categoryContainer}>
      <div className={styles.category}>
        {name}
      </div>
      <DisadvantageDot isDisadvantaged={isDisadvantaged}/>
    </div>
  );
};
export default Category;
