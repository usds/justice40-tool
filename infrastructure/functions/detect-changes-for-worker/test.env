REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1