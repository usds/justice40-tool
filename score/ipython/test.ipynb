{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
