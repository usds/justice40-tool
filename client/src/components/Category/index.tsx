import Category from './Category';

export default Category;
