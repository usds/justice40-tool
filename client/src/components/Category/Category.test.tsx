import * as React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import Category from './Category';

describe('rendering of the Category disadvantaged', () => {
  const {asFragment} = render(
      <LocalizedComponent>
        <Category name={'test name'} isDisadvantaged={true}/>
      </LocalizedComponent>,
  );

  it('checks if component renders', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('rendering of the Category non-disadvantaged', () => {
  const {asFragment} = render(
      <LocalizedComponent>
        <Category name={'test name'} isDisadvantaged={false}/>
      </LocalizedComponent>,
  );

  it('checks if component renders', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
