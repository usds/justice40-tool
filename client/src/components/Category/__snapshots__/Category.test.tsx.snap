// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of the Category disadvantaged checks if component renders 1`] = `
<DocumentFragment>
  <div>
    <div>
      test name
    </div>
    <div />
  </div>
</DocumentFragment>
`;

exports[`rendering of the Category non-disadvantaged checks if component renders 1`] = `<DocumentFragment />`;
